ui-admin/src/study/CreateNewCohortModal.tsx

```diff
@@ -0,0 +1,124 @@
+import { useNavContext } from '../navbar/NavContextProvider'
+import React, { useEffect, useState } from 'react'
+import { Portal, PortalStudy } from '@juniper/ui-core'
+import Modal from 'react-bootstrap/Modal'
+import { Button } from '../components/forms/Button'
+import Select from 'react-select'
+import useReactSingleSelect from '../util/react-select-utils'
+import Api, { ExportData } from '../api/api'
+import { useLoadingEffect } from 'api/api-utils'
+import { Field, QueryBuilder } from 'react-querybuilder'
+import 'react-querybuilder/dist/query-builder.css'
+import { QueryBuilderBootstrap } from '@react-querybuilder/bootstrap'
+import { Store } from 'react-notifications-component'
+import { failureNotification } from '../util/notifications'
+
+/**
+ * Returns a cohort builder modal
+ */
+export default function CreateNewCohortModal({ onDismiss }: {onDismiss: () => void}) {
+  const { portalList } = useNavContext()
+  const initialQuery = { combinator: 'and', rules: [] }
+  const [query, setQuery] = useState(initialQuery)
+  const [cohortName, setCohortName] = useState('')
+  const [selectedPortal, setSelectedPortal] = useState<Portal>()
+  const [selectedStudy, setSelectedStudy] = useState<PortalStudy>()
+  const [participantFields, setParticipantFields] = useState<ExportData>()
+
+  useLoadingEffect(async () => {
+    if (!selectedStudy || !selectedPortal) { return }
+    const response = await Api.exportEnrollees(
+      selectedPortal.shortcode,
+      selectedStudy.study.shortcode,
+      'live', { fileFormat: 'JSON', limit: 0 })
```

Comment:
This seemed like a quick and easy way to get all of the fields that someone might want to match on. In the future we'll probably want a dedicated endpoint for populating the field selector that also returns the field datatype. Then we can customize the valueEditor portion of the react-querybuilder to have better handling for booleans (checkbox), dates (date picker), etc. Right now it just treats everything as strings.